core/mock/mock.go

```diff
@@ -54,6 +54,7 @@ func NewMockNode() (*core.IpfsNode, error) {
 	nd.Peerstore = peer.NewPeerstore()
 	nd.Peerstore.AddPrivKey(p, ident.PrivateKey())
 	nd.Peerstore.AddPubKey(p, ident.PublicKey())
+	nd.Identity = p
```

Comment:
@rht you're right, that's a bug. `core.Offline(...)` should initialize from the `config.Identity` -- though maybe the problem is a lack of a private key? not sure.